src/sagas/wallet.js

```diff
@@ -114,12 +114,11 @@ export function* startWallet(action) {
   yield put(loadingAddresses(true));
   yield put(storeRouterHistory(routerHistory));
 
-  const walletId = yield LOCAL_STORE.getWalletId();
-  if (!walletId) {
-    // The wallet has not been initialized yet
-    if (hardware) {
-      yield LOCAL_STORE.initHWStorage(xpub);
-    } else {
+  if (hardware) {
+    yield LOCAL_STORE.initHWStorage(xpub);
+  } else {
+    const walletId = yield LOCAL_STORE.getWalletId();
+    if (!walletId) {
```

Comment:
Here, we always init the hardware storage, this will ensure we generate the wallet id from the xpub from scratch to avoid issues when using 2 different ledger devices.

Reply:
Can you add this comment to the code, please?

Reply:
Done